rust/worker/src/segment/types.rs

```diff
@@ -112,20 +112,25 @@ impl ChromaError for LogMaterializerError {
     }
 }
 
-#[derive(Debug, Clone)]
-pub struct MaterializedLogRecord<'referred_data> {
+/// This struct is used internally. It is not exposed to materialized log consumers.
+///
+/// Instead of cloning or holding references to log records/segment data, this struct contains owned values that can be resolved to the referenced data.
+/// E.x. `data_record_offset_id: Option<u32>` is used instead of `data_record: Option<&DataRecord>` to avoid holding references to the data.
+/// This allows `MaterializedLogRecord` (and types above it) to be trivially Send'able.
```

Comment:
nit: I think this has more to do with the 'static bound than the `Send` bound

Reply:
two sides of the same coin I think, iiuc a struct with stored &'static references will be `Send`/`Sync` but that doesn't semantically make sense